File: Models/campbell_SoilTemp/Campbell/src/java/Campbell/campbellExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class campbellExogenous
{
    private double SRAD;
    private double TMIN;
    private Integer DOY;
    private Double [] SW;
    private double EOAD;
    private double ESP;
    private double TAV;
    private double ESAD;
    private double canopyHeight;
    private double TMAX;
    private double ES;
    
    public campbellExogenous() { }
    
    public campbellExogenous(campbellExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.SRAD = toCopy.getSRAD();
            this.TMIN = toCopy.getTMIN();
            this.DOY = toCopy.getDOY();
            SW = new Double[toCopy.getSW().length];
        for (int i = 0; i < toCopy.getSW().length; i++)
        {
            SW[i] = toCopy.getSW()[i];
        }
            this.EOAD = toCopy.getEOAD();
            this.ESP = toCopy.getESP();
            this.TAV = toCopy.getTAV();
            this.ESAD = toCopy.getESAD();
            this.canopyHeight = toCopy.getcanopyHeight();
            this.TMAX = toCopy.getTMAX();
            this.ES = toCopy.getES();
        }
    }
    public double getSRAD()
    { return SRAD; }

    public void setSRAD(double _SRAD)
    { this.SRAD= _SRAD; } 
    
    public double getTMIN()
    { return TMIN; }

    public void setTMIN(double _TMIN)
    { this.TMIN= _TMIN; } 
    
    public Integer getDOY()
    { return DOY; }

    public void setDOY(Integer _DOY)
    { this.DOY= _DOY; } 
    
    public Double [] getSW()
    { return SW; }

    public void setSW(Double [] _SW)
    { this.SW= _SW; } 
    
    public double getEOAD()
    { return EOAD; }

    public void setEOAD(double _EOAD)
    { this.EOAD= _EOAD; } 
    
    public double getESP()
    { return ESP; }

    public void setESP(double _ESP)
    { this.ESP= _ESP; } 
    
    public double getTAV()
    { return TAV; }

    public void setTAV(double _TAV)
    { this.TAV= _TAV; } 
    
    public double getESAD()
    { return ESAD; }

    public void setESAD(double _ESAD)
    { this.ESAD= _ESAD; } 
    
    public double getcanopyHeight()
    { return canopyHeight; }

    public void setcanopyHeight(double _canopyHeight)
    { this.canopyHeight= _canopyHeight; } 
    
    public double getTMAX()
    { return TMAX; }

    public void setTMAX(double _TMAX)
    { this.TMAX= _TMAX; } 
    
    public double getES()
    { return ES; }

    public void setES(double _ES)
    { this.ES= _ES; } 
    
}

File: Models/campbell_SoilTemp/Campbell/src/cs/Campbell/campbellExogenous.cs
using System;
using System.Collections.Generic;

public class CampbellExogenous 
{
    private double _ESP;
    private double _windSpeed;
    private int _DOY;
    private double[] _SW;
    private double _canopyHeight;
    private double _TMIN;
    private double _TMAX;
    private double _TAV;
    private double _ES;
    private double _EOAD;
    private double _T2M;
    private double _SRAD;
    
    /// <summary>
    /// Constructor of the campbellExogenous component")
    /// </summary>  
    public campbellExogenous() { }
    
    
    public campbellExogenous(campbellExogenous toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            ESP = toCopy.ESP;
            windSpeed = toCopy.windSpeed;
            DOY = toCopy.DOY;
            SW = new double[toCopy.SW.Length];
            for (int i = 0; i < toCopy.SW.Length; i++)
                { SW[i] = toCopy.SW[i]; }
    
            canopyHeight = toCopy.canopyHeight;
            TMIN = toCopy.TMIN;
            TMAX = toCopy.TMAX;
            TAV = toCopy.TAV;
            ES = toCopy.ES;
            EOAD = toCopy.EOAD;
            T2M = toCopy.T2M;
            SRAD = toCopy.SRAD;
        }
    }
    public double ESP
    {
        get { return this._ESP; }
        set { this._ESP= value; } 
    }
    public double windSpeed
    {
        get { return this._windSpeed; }
        set { this._windSpeed= value; } 
    }
    public int DOY
    {
        get { return this._DOY; }
        set { this._DOY= value; } 
    }
    public double[] SW
    {
        get { return this._SW; }
        set { this._SW= value; } 
    }
    public double canopyHeight
    {
        get { return this._canopyHeight; }
        set { this._canopyHeight= value; } 
    }
    public double TMIN
    {
        get { return this._TMIN; }
        set { this._TMIN= value; } 
    }
    public double TMAX
    {
        get { return this._TMAX; }
        set { this._TMAX= value; } 
    }
    public double TAV
    {
        get { return this._TAV; }
        set { this._TAV= value; } 
    }
    public double ES
    {
        get { return this._ES; }
        set { this._ES= value; } 
    }
    public double EOAD
    {
        get { return this._EOAD; }
        set { this._EOAD= value; } 
    }
    public double T2M
    {
        get { return this._T2M; }
        set { this._T2M= value; } 
    }
    public double SRAD
    {
        get { return this._SRAD; }
        set { this._SRAD= value; } 
    }
}